src/ConnectionManager.ts

```diff
@@ -91,12 +91,35 @@ export class ConnectionManager extends EventEmitter {
         throw new ApiError(`Connection type not known`);
     }
 
-    private assertStateAllowed(state: StateEvent<any>, serviceType: string) {
-        if (state.sender === this.as.botUserId) {
-            return;
+    public assertStateAllowed(roomId: string, state: StateEvent, serviceType: string) {
+        if (!this.isStateAllowed(roomId, state, serviceType)) {
+            this.tryRestoreState(roomId, state, serviceType);
```

Comment:
Personal preference: Tell the next person that you don't want to `catch` errors of an async function.

```suggestion
            void this.tryRestoreState(roomId, state, serviceType);
```

Reply:
Ah, I didn't know about [the `void` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void), thanks!